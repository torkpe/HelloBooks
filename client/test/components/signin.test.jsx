import React from 'react';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import sinon from 'sinon';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import storeConfiguration from '../../store/store';
import ConnectedSigninPage, { Signin } from '../../components/Signin.jsx';

configure({ adapter: new Adapter() });

const props = {
  userSignin: jest.fn(),
  clearSigninState: jest.fn(),
  signin: {
    isLoading: false,
    successfullySignedin: {
      myToken: 'yggjbhbjbhbjbhjbhjbhjvbhjvvhjvh'
    },
    errors: {
      message: 'mbnmn m,mm'
    },
  }
};
const event = {
  preventDefault: jest.fn(),
  target: {
    username: 'okon',
    password: 'abcde',
    email: 'okon@yahoo.com',
  }
};
const initialState = {
  userSignin: {
    successfullySignedin: {
      myToken: 'yggjbhbjbhbjbhjbhjbhjvbhjvvhjvh'
    },
    isLoading: false,
    errors: {
      message: 'mbnmn m,mm'
    },
  }
};
const mockStore = configureMockStore();
let mountedComponent;
let store;
let wrapper;
let shallowComponent;

describe('<Signin />', () => {
  beforeEach(() => {
    store = mockStore(storeConfiguration());
    mountedComponent = mount(<Signin {...props} />);
    shallowComponent = shallow(<Signin {...props} />);
  });
  it('renders <Signin /> component', () => {
    const tree = toJson(mountedComponent);
    expect(tree).toMatchSnapshot();
  });
  it('calls componentWillReceiveProps', () => {
    sinon.spy(Signin.prototype, 'componentWillReceiveProps');
    shallowComponent.instance().componentWillReceiveProps(props);
    expect(Signin.prototype.componentWillReceiveProps.calledOnce).toEqual(true);
  });
  it('calls componentWillReceiveProps when there is no token', () => {
    props.signin.successfullySignedin.myToken = null;
    shallowComponent.instance().componentWillReceiveProps(props);
    expect(Signin.prototype.componentWillReceiveProps.calledOnce).toEqual(false);
  });
  it('calls componentWillReceiveProps when there is no error message', () => {
    props.signin.errors.message = null;
    shallowComponent.instance().componentWillReceiveProps(props);
    expect(Signin.prototype.componentWillReceiveProps.calledOnce).toEqual(false);
  });
  it('calls handle Signup event after login button is clicked', () => {
    sinon.spy(Signin.prototype, 'onSubmit');
    props.signin.successfullySignedin.myToken = 'mvhbvhjbbbb';
    shallowComponent = shallow(<Signin {...props} />);
    shallowComponent.instance().onSubmit(event);
    expect(Signin.prototype.onSubmit.calledOnce).toEqual(true);
  });
  it('calls handleChange event', () => {
    sinon.spy(Signin.prototype, 'onChange');
    shallowComponent.instance().onChange(event);
    expect(Signin.prototype.onChange.calledOnce).toEqual(false);
  });
});
