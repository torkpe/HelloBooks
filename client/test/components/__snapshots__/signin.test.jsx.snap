// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin /> renders <Signin /> component 1`] = `
<Signin
  clearSigninState={[MockFunction]}
  signin={
    Object {
      "errors": Object {
        "message": "mbnmn m,mm",
      },
      "isLoading": false,
      "successfullySignedin": Object {
        "myToken": "yggjbhbjbhbjbhjbhjbhjvbhjvvhjvh",
      },
    }
  }
  userSignin={[MockFunction]}
>
  <div
    className="mdl-grid"
  >
    <div
      className="contents"
    >
      <div
        className="card-enlarge form-card mdl-card mdl-shadow--3dp"
      >
        <form
          onSubmit={[Function]}
        >
          <h5>
             Sign in
          </h5>
          <EmailForm
            onChange={[Function]}
          >
            <div
              className="card-content card-wrapper input-wrapper"
            >
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
              />
            </div>
          </EmailForm>
          <div
            className="card-content card-wrapper input-wrapper"
          >
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <button
            className="
    mdl-button
    mdl-js-button
    mdl-button--raised
    mdl-button--colored"
            disabled={false}
            id="button"
            name="signin"
          >
            Sign in
          </button>
        </form>
        <div
          className="mdl-card__supporting-text"
        >
          Don't Have an Account yet? Create one Below
        </div>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/signup"
        />
        <div
          className="mdl-card__supporting-text"
        >
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/forgot-password"
          />
        </div>
      </div>
    </div>
  </div>
</Signin>
`;
